k8s/upgrade/src/bin/upgrade-job/upgrade.rs

```diff
@@ -37,20 +37,13 @@ pub(crate) async fn upgrade(opts: &CliArgs) -> Result<()> {
         .build()
         .await?;
 
-    let helm_upgrade_dry_run = HelmUpgrade::builder()
-        .with_namespace(opts.namespace())
-        .with_release_name(opts.release_name())
-        .with_core_chart_dir(opts.core_chart_dir())
-        .with_skip_upgrade_path_validation(opts.skip_upgrade_path_validation())
-        .with_values(opts.values())
-        .build()
-        .await?;
-
-    if let Err(error) = helm_upgrade_dry_run.dry_run().await {
+    // Dry-run helm upgrade.
+    let dry_run_result: Result<HelmUpgradeRunner> = helm_upgrade.dry_run().await;
+    if let Err(error) = dry_run_result {
         event.publish_unrecoverable(&error, true).await;
-        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
         return Err(error);
     }
+    let run_helm_upgrade = dry_run_result.unwrap();
```

Comment:
```suggestion
    let run_helm_upgrade = match dry_run_result {
        Err(error) => {
            event.publish_unrecoverable(&error, true).await;
            Err(error)
        }
        Ok(run_helm_upgrade) => Ok(run_helm_upgrade)
    }?;
```

Reply:
This doesn't return the error. Just assigns it. We'd have to add a return?

Reply:
Yes it does, through the `?`